lib/heap_sort.rb

```diff
@@ -1,8 +1,22 @@
-
+require_relative 'min_heap'
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  o((n * log n) + (n * log n)) => o(n log n) | it depends on the length of the list, but adding and removing from the list takes log(n), and the removal of each takes log(n) as well, and has to happen n times (the length of the list)
+# Space Complexity: also o(n log n)
+def heapsort(list)
```

Comment:
The method here works, great!  However the space complexity is O(n) because you're making a heap of n elements, not n log n elements.